service/lib/dinstaller/software/manager.rb

```diff
@@ -112,10 +116,17 @@ def propose
         proposal = Yast::Packages.Proposal(force_reset = false, reinit = false, _simple = true)
         logger.info "proposal #{proposal["raw_proposal"]}"
 
-        solve_dependencies
+        deps_result = solve_dependencies
 
-        # do not return proposal hash, so intentional nil here
-        nil
+        proposal_result(proposal, deps_result)
+      end
+
+      def validate
+        # validation without probing does not make sense and produces false errors
+        return [] unless @probed
+
+        msgs = propose
```

Comment:
I would say `#propose` is still performed with each call to `#validate`.

Reply:
well, it is only for validation called after probing. In that case something should trigger it, as validation has its properties and only if something invalidate it, it is recomputed. And it is invalidated from software proposal which change required packages or after change of product. So when I test it, I do not see unnecessary calls to validate as it is already cached and I need to reset it appropriate places with `update_validation`. So in fact my issue was to reset cache and not to cache more.

Reply:
Ok, I see. The only problem I see is when calling to `software.manager.propose` D-Bus method. In that case, the proposal is done twice, right?

Reply:
yeap, I do not notice it much, but should check logs for timestamp, so we have idea how long it take.